### PR TITLE
Attach 'hide workbench' button to workbench element

### PR DESCRIPTION
**Pull request by chloeleichen**
Fixing:https://github.com/TerriaJS/terriajs/issues/2983

Also added a `Responsive.jsx` React Component to handle media query in this PR, as it is used in `StandardUserInterface`. This is not completed in other files and will be further address in https://github.com/TerriaJS/terriajs/issues/3061. Can take that bit out if you think it's more appropriate.

### CODE REVIEW

**Comment by kring** [comment]
Looks like there are some issues in 2D:
image

File: lib/ReactViews/Map/MenuBar.jsx [hunk 1]
```diff
@@ -1,21 +1,20 @@
-import React from 'react';
+import React from "react";
```
Comment (kring) [inline lib/ReactViews/Map/MenuBar.jsx]: Still using single quotes in terria, sorry!

**Comment by kring** [comment]
There's a problem with the "medium" size - no workbench:
image

File: lib/ReactViews/Generic/Responsive.jsx [hunk 2]
```diff
@@ -0,0 +1,43 @@
+import React from "react";
+import MediaQuery from "react-responsive";
+import PropTypes from "prop-types";
+
+// This should come from some config some where
+const small = 768;
+const medium = 992;
+const large = 1200;
```
Comment (kring) [inline lib/ReactViews/Generic/Responsive.jsx]: `_variables.scss` says `large` is 1300, not 1200. Is this a mistake?

File: lib/ReactViews/SidePanel/FullScreenButton.jsx [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+            <div
+                className={classNames(Styles.fullScreen, {
+                    [Styles.minifiedFullscreenBtnWrapper]: this.props.minified
+                })}
+            >
+                <button
+                    type="button"
+                    onClick={this.toggleFullScreen}
+                    className={btnClassName}
+                    title="Toggle full screen"
```
Comment (kring) [inline lib/ReactViews/SidePanel/FullScreenButton.jsx]: Unless Phil has said otherwise, I think this should say "Hide workbench" or "Show workbench" instead of "Toggle full screen". Full screen is a different thing.

File: lib/ReactViews/SidePanel/SidePanel.jsx [hunk 4]
```diff
@@ -78,46 +81,83 @@ const SidePanel = createReactClass({
...
+                        >
+                            <Icon glyph={Icon.GLYPHS.add} />Add data
                         </button>
                     </div>
                 </div>
                 <div className={Styles.body}>
                     <Choose>
-                        <When condition={searchState.locationSearchText.length > 0 && searchState.showLocationSearchResults}>
+                        <When
+                            condition={
```
Comment (kring) [inline lib/ReactViews/SidePanel/SidePanel.jsx]: Lots of unnecessary formatting changes here making it hard to see what's actually changed.

**Comment by kring** [comment]
Thanks @chloeleichen, looking good. Please don't use prettier in Terria, though. If we're going to start using it, we need to do it all at once in a separate PR, cause otherwise diffs are impossible to follow.

**Comment by chloeleichen** [comment]
@kring Thanks, reverted most format changes. fixed the issue with median screen size and button title. 💃

File: lib/ReactViews/SidePanel/FullScreenButton.jsx [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+            <div
+                className={classNames(Styles.fullScreen, {
+                    [Styles.minifiedFullscreenBtnWrapper]: this.props.minified
+                })}
+            >
+                <button
+                    type="button"
+                    onClick={this.toggleFullScreen}
+                    className={btnClassName}
+                    title={this.props.viewState.isMapFullScreen ? 'Hide full screen' : 'Show full screen'}
```
Comment (kring) [inline lib/ReactViews/SidePanel/FullScreenButton.jsx]: Show/hide Workbench, please. :)

File: lib/ReactViews/StandardUserInterface/StandardUserInterface.jsx [hunk 6]
```diff
@@ -1,31 +1,33 @@
...
-import Notification from './../Notification/Notification.jsx';
-import ObserveModelMixin from './../ObserveModelMixin';
-import ProgressBar from '../Map/ProgressBar.jsx';
-import SidePanel from './../SidePanel/SidePanel.jsx';
-import processCustomElements from './processCustomElements';
-import classNames from 'classnames';
-import 'inobounce';
-
-import Styles from './standard-user-interface.scss';
+import React from "react";
```
Comment (kring) [inline lib/ReactViews/StandardUserInterface/StandardUserInterface.jsx]: Quotes and formatting in this file too, sorry!

**Comment by kring** [comment]
Looks like the build is failing because of an eslint warning.

File: lib/ReactViews/SidePanel/FullScreenButton.jsx (hunk 5 above)
Comment (chloeleichen) [reply lib/ReactViews/SidePanel/FullScreenButton.jsx]: oops sorry 👍

**Comment by kring** [comment]
Thanks @chloeleichen!
